lib/csu/aarch64c/reloc.c

```diff
@@ -60,6 +60,7 @@ init_cap_from_fragment(const Elf_Addr *fragment, void * __capability data_cap,
 		cap = cheri_setbounds(cap, len);
 	}
 
+	cap = cheri_clearperm(cap, CHERI_PERM_SW_VMEM);
```

Comment:
I would either hoist this to be up with the cheri_setaddress or fold this into the existing cheri_clearperm cases

Reply:
Hoisted up, outside conditionals to emphasize that the clearing is non-conditional.